### PR TITLE
[posix] add microsecond timer support

### PR DESCRIPTION
**Pull request by chshu**

### CODE REVIEW

File: examples/platforms/posix/alarm.c [hunk 1]
```diff
@@ -32,83 +32,145 @@
...
     if (aTimeout == NULL)
     {
         return;
     }
 
-    if (s_is_running)
+    if (sIsUsRunning)
     {
-        remaining = (int32_t)(s_alarm - otPlatAlarmMilliGetNow());
+        remaining = (int32_t)(sUsAlarm - otPlatAlarmMicroGetNow());
```
Comment (bukepo) [inline examples/platforms/posix/alarm.c]: Is it simpler to compare `remaining` values of milliseconds and microseconds?

Comment (chshu) [inline examples/platforms/posix/alarm.c]: Updated, thanks for the suggestion.

File: examples/platforms/posix/openthread-core-posix-config.h [hunk 2]
```diff
@@ -51,4 +51,12 @@
  */
 #define OPENTHREAD_CONFIG_ENABLE_DEFAULT_LOG_OUTPUT             1
 
+/**
+ * @def OPENTHREAD_CONFIG_ENABLE_PLATFORM_USEC_TIMER
+ *
+ * Define to 1 if you want to support microsecond timer in platform.
+ *
+ */
+#define OPENTHREAD_CONFIG_ENABLE_PLATFORM_USEC_TIMER     1
```
Comment (jwhui) [inline examples/platforms/posix/openthread-core-posix-config.h]: nit: align value with those above

Comment (chshu) [inline examples/platforms/posix/openthread-core-posix-config.h]: Done, thanks.

**Pull request merged**